firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java

```diff
@@ -242,6 +242,7 @@ private boolean shouldIgnoreCache(String event) {
                                           Locale.US,
                                           "Successfully fetched %d messages from backend",
                                           resp.getMessagesList().size())))
+                          .doOnSuccess(impressionStorageClient::clearImpressions)
```

Comment:
`.doOnSuccess(s -> impressionStorageClient.clearImpressions(s).subscribe())` similar to what cacheWrite does at line 266.

The key here is that doOnSuccess is a side effect type operator. The execution framework assumes that nothing needs to happen with the returned value. In your case, the return type of clearImpressions is a `Completable` that needs to either get chained into the observable or completed within the success consumer.